src/gas.rs

```diff
@@ -640,4 +808,225 @@ mod tests {
 
 	}
 
+	#[test]
+	fn branch_innermost() {
```

Comment:
I find these new tests harder to read compared to diff tests. Should we move them to be test fixtures?

Reply:
Yeah... I prefer these small, targeted examples as unit tests but certainly the syntax is difficult to read. I'll write a macro so that we can write cleaner unit tests like these in WAT.

Reply:
I won't insist on this. Check out https://github.com/paritytech/wasm-utils/blob/c3d10a26196b34a5041328fa7a5916bd53bd0350/src/stack_height/mod.rs#L396 cause it uses wat2wasm which can help here